src/app/favoritePlaces/favorite.controller.js

```diff
@@ -0,0 +1,97 @@
+(function() {
+    'use strict';
+
+    angular
+        .module('project')
+        .controller('FavoritePlacesController', FavoritePlacesController);
+
+    FavoritePlacesController.$inject = ['$scope', 'FacebookService', 'userService', 'googleMapPositionService', 'cacheService', '$localStorage', '$sessionStorage', 'errorHandling'];
+
+    function FavoritePlacesController($scope, FacebookService, userService, googleMapPositionService, cacheService, $localStorage, $sessionStorage, errorHandling) {
+        var favoriteVm = this;
+        
+        favoriteVm.$storage = $localStorage;
+        favoriteVm.favoriteList = favoriteVm.$storage.items;
+        favoriteVm.user = userService.user;
+        favoriteVm.removeFavorite = removeFavorite;
+        favoriteVm.ready = false;
+        favoriteVm.googleMapApiReady;
+        favoriteVm.state;
+        favoriteVm.onDropComplete = onDropComplete;
+        favoriteVm.listOfObjectPlaces = [];
+        favoriteVm.position = {};
+
+        activate();
+
+        function activate() {
+            FacebookService.checkLoginState()
+                .then(function(){
+                    favoriteVm.ready = true;
+                    initGoogleMapApi();
+                    checkFBState();
```

Comment:
Too much checking the FB state :> lets replace this line with line 69, then remove checkFBState function :)